data-protocols/dsp/dsp-catalog/dsp-catalog-api/src/main/java/org/eclipse/edc/protocol/dsp/catalog/api/controller/CatalogController.java

```diff
@@ -0,0 +1,119 @@
+import jakarta.json.JsonObject;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.HeaderParam;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.MediaType;
+import org.eclipse.edc.catalog.spi.Catalog;
+import org.eclipse.edc.jsonld.transformer.JsonLdTransformerRegistry;
+import org.eclipse.edc.protocol.dsp.catalog.spi.types.CatalogRequestMessage;
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.iam.IdentityService;
+import org.eclipse.edc.spi.iam.TokenRepresentation;
+import org.eclipse.edc.web.spi.exception.AuthenticationFailedException;
+import org.eclipse.edc.web.spi.exception.InvalidRequestException;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static org.eclipse.edc.jsonld.util.JsonLdUtil.compact;
+import static org.eclipse.edc.jsonld.util.JsonLdUtil.expand;
+import static org.eclipse.edc.protocol.dsp.catalog.spi.CatalogApiPaths.BASE_PATH;
+import static org.eclipse.edc.protocol.dsp.catalog.spi.CatalogApiPaths.CATALOG_REQUEST;
+import static org.eclipse.edc.protocol.dsp.catalog.transform.DspCatalogPropertyAndTypeNames.DSPACE_PREFIX;
+import static org.eclipse.edc.protocol.dsp.catalog.transform.DspCatalogPropertyAndTypeNames.DSPACE_SCHEMA;
+import static org.eclipse.edc.protocol.dsp.transform.transformer.Namespaces.DCAT_PREFIX;
+import static org.eclipse.edc.protocol.dsp.transform.transformer.Namespaces.DCAT_SCHEMA;
+import static org.eclipse.edc.protocol.dsp.transform.transformer.Namespaces.DCT_PREFIX;
+import static org.eclipse.edc.protocol.dsp.transform.transformer.Namespaces.DCT_SCHEMA;
+import static org.eclipse.edc.protocol.dsp.transform.transformer.Namespaces.ODRL_PREFIX;
+import static org.eclipse.edc.protocol.dsp.transform.transformer.Namespaces.ODRL_SCHEMA;
+
+/**
+ * Provides the HTTP endpoint for receiving catalog requests.
+ */
+@Consumes({MediaType.APPLICATION_JSON})
+@Produces({MediaType.APPLICATION_JSON})
+@Path(BASE_PATH)
+public class CatalogController {
+    
+    private ObjectMapper mapper;
+    private IdentityService identityService;
+    private JsonLdTransformerRegistry transformerRegistry;
+    private String dspCallbackAddress;
+    
+    public CatalogController(ObjectMapper mapper, IdentityService identityService,
+                             JsonLdTransformerRegistry transformerRegistry, String dspCallbackAddress) {
+        this.mapper = mapper;
+        this.identityService = identityService;
+        this.transformerRegistry = transformerRegistry;
+        this.dspCallbackAddress = dspCallbackAddress;
+    }
+    
+    @POST
+    @Path(CATALOG_REQUEST)
+    public Map<String, Object> getCatalog(JsonObject jsonObject, @HeaderParam(HttpHeaders.AUTHORIZATION) String token) {
+        var expanded = expand(jsonObject).getJsonObject(0); //expanding returns a JsonArray of size 1
```

Comment:
Is there a specific reason why this "expansion" operation is not in the transformer?

Reply:
Could probably also be done in the transformers, if we want them to return the compacted form. As I built this up on Jim's initial proposal for the transformer structure, @jimmarino what's your opinion on this?

But as this would also entail refactoring all the transformer tests, I would vote to change this in a separate PR, would then open a respective issue.

Reply:
I think the expansion should be done outside the transformers because it needs to be performed at the root of the object graph, and the catalog is not necessarily the root in all foreseeable cases. For example, it may be contained in another structure or its elements may be top-level items in different contexts.